src/source_clang.cc

```diff
@@ -1112,18 +1112,16 @@ Source::ClangViewAutocomplete(file_path, language) {
     return usages;
   };
   
-  goto_method=[this](){
+  get_methods=[this](){
+    std::vector<std::pair<Offset, std::string> > methods;
     if(!parsed) {
       Info::get().print("Buffer is parsing");
-      return;
+      return methods;
     }
-    auto iter=get_iter_for_dialog();
-    selection_dialog=std::unique_ptr<SelectionDialog>(new SelectionDialog(*this, get_buffer()->create_mark(iter), true, true));
-    auto rows=std::make_shared<std::unordered_map<std::string, clang::Offset> >();
-    auto methods=clang_tokens->get_cxx_methods();
-    if(methods.size()==0)
-      return;
-    for(auto &method: methods) {
+    auto cxx_methods=clang_tokens->get_cxx_methods();
+    if(cxx_methods.size()==0)
+      return methods;
```

Comment:
I think this check is redundant, the for-loop below does essentially the same thing. I realise this wasn't added by this commit though.